### PR TITLE
tests:chore - move Mock implementations to testutil pkg

### PR DESCRIPTION
**Pull request by matheusalcantarazup**
Previously each mock implementation of structs that is used by tests was
implemented on your own package, which add some unnecessary testing
dependencies on main packages.

This commit move all mocks implementation to testutil package and change
their references.

Note that the `internal/utils/mock` package was also removed and their
code was moved to testutil package to centralize all testing stuffs
only on testutil package.

Signed-off-by: Matheus Alcantara <matheus.alcantara@zup.com.br>

**- What I did**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Review by wiliansilvazup — CHANGES_REQUESTED** [review]
to not get too confused, what do you think of having a file split of "mock", "e2e tests" "unit tests" and others? or doesn't it make sense?

**Comment by matheusalcantarazup** [comment]
> to not get too confused, what do you think of having a file split of "mock", "e2e tests" "unit tests" and others? or doesn't it make sense?

Are you saying for example use `testutil/requirements_mock.go` instead? I think that would be good.

About unit tests and e2e I don't think soo. Some funcs and const can be used in both tests.

**Review by iancardosozup — APPROVED** [review]

File: internal/utils/testutil/request_mock.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	"net/http"
+
+	"github.com/ZupIT/horusec-devkit/pkg/services/http/request/entities"
+
+	"github.com/stretchr/testify/mock"
+
+	mockutils "github.com/ZupIT/horusec-devkit/pkg/utils/mock"
+)
+
+type RequestMock struct {
```
[the remark this answers is not in the document]
Comment (matheusalcantarazup) [reply internal/utils/testutil/request_mock.go:28]: This is good!
We should remove this `RequestMock` implementation and change their reference or make an alias `type RequestMock request.Mock`? What do you think?

[the remark this answers is not in the document]
Comment (nathanmartinszup) [reply internal/utils/testutil/request_mock.go:28]: Any of the solutions sound good to me, maybe the alias to keep all in the same place.

**Review by nathanmartinszup — APPROVED** [review]

**Comment by wiliansilvazup** [comment]
> > to not get too confused, what do you think of having a file split of "mock", "e2e tests" "unit tests" and others? or doesn't it make sense?
> 
> Are you saying for example use `testutil/requirements_mock.go` instead? I think that would be good.
> 
> About unit tests and e2e I don't think soo. Some funcs and const can be used in both tests.

oh great :)

**Review by wiliansilvazup — APPROVED** [review]

**Comment by wiliansilvazup** [comment]
sync your branch with main and you can merge this pr :)

**Pull request merged**